scripts/helpers/customBuild.js

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const fs = require("fs");
+const { rollup } = require("rollup");
+const nodeResolve = require("@rollup/plugin-node-resolve").default;
+const commonjs = require("@rollup/plugin-commonjs");
+const babel = require("@rollup/plugin-babel").default;
+const terser = require("rollup-plugin-terser").terser;
+const yargs = require("yargs/yargs");
+const { hideBin } = require("yargs/helpers");
+const componentNames = require("./componentNames");
+const conditionalBuildBabelPlugin = require("./conditionalBuildBabelPlugin");
+
+const untouchableComponent = [
+  "context",
+  "privacy",
+  "identity",
+  "datacollector",
+  "libraryinfo"
+];
```

Comment:
I don't think any component should be "untouchable". If there is some logic that is untouchable it should be incorporated into core. Some of these components will need some work to make that a reality.

- context - This could be removed except for the fact that it handles adding a timestamp and implementationInfo to the XDM. This functionality should be put into core.
- privacy - I believe you can leave this out now without any problems. I see this being a big use case for customers too if they don't ever use consent.
- identity - I can see a customer wanting to remove this if they use FPID. I think the library would work without identity right now.
- datacollector - Currently this component defines "sendEvent" command. We should probably make "sendEvent" command part of core, and then each component can apply its own logic via onBeforeEvent lifecycle hook.
- libraryinfo - We could make this part of core.

Reply:
@jonsnyder Do you think we should do this before carrying on with the custom build task?